src/domain/Processors/VehicleProcessor.ts

```diff
@@ -0,0 +1,533 @@
+    vehicle.partialVin = vehicle.partialVin?.toUpperCase();
+    vehicle.primaryVrm = vehicle.primaryVrm?.toUpperCase();
+    vehicle.secondaryVrms = vehicle.secondaryVrms?.map((vrm: string) =>
+      vrm.toUpperCase()
+    );
+    return vehicle;
+  }
+  /* #endregion */
+
+  /* #region  Public functions */
+  /**
+   * validates and inserts a new tech record in the database
+   * @param msUserDetails the user who created the tech record
+   */
+  public async createVehicle(msUserDetails: IMsUserDetails): Promise<T> {
+    try {
+      this.vehicle.techRecord[0] = this.validate(this.vehicle, true);
+      await this.setNumberKey();
+      this.vehicle.techRecord[0] = this.mapFields(this.vehicle.techRecord[0]);
+      this.vehicle = this.capitaliseGeneralVehicleAttributes(this.vehicle);
+      AuditDetailsHandler.setAuditDetailsAndStatusCodeForNewRecord(
+        this.vehicle.techRecord[0],
+        msUserDetails
+      );
+      this.vehicle.techRecord[0].recordCompleteness = computeRecordCompleteness(
+        this.vehicle
+      );
+      const data = await this.techRecordDAO.createSingle(this.vehicle);
+      return data.Attributes as T;
+    } catch (error) {
+      console.error(error);
+      const errorList = error.body?error.body.errors:error.message;
+      throw ErrorHandler.Error(error.statusCode, errorList);
+    }
+  }
+
+  /**
+   * Validate and update tech record
+   * @param msUserDetails the user who has made changes to the record.
+   * @param oldStatusCode the old status code of the tech record. It will be passed if the user changes this value.
+   */
+  public async updateVehicle(
+    msUserDetails: IMsUserDetails,
+    oldStatusCode?: STATUS
+  ) {
+    try {
+      const data = await this.createAndArchiveTechRecord(
+        this.vehicle,
+        msUserDetails,
+        oldStatusCode
+      );
+      const updatedData = await this.techRecordDAO.updateSingle(data);
+      return this.techRecordsListHandler.formatTechRecordItemForResponse(
+        updatedData.Attributes as T
+      );
+    } catch (error) {
+      console.error(error);
+      throw ErrorHandler.Error(error.statusCode, error.body.errors);
+    }
+  }
+
+  public async addNewProvisionalRecord(
+    msUserDetails: IMsUserDetails
+  ): Promise<T> {
+    if (this.vehicle.techRecord[0].statusCode !== STATUS.PROVISIONAL) {
+      throw ErrorHandler.Error(400, ERRORS.STATUS_CODE_SHOULD_BE_PROVISIONAL);
+    }
+
+    try {
+      this.validate(this.vehicle, false);
+      const vehiclesFromDB = await this.techRecordsListHandler.getFormattedTechRecordsList(
+        this.vehicle.systemNumber,
+        STATUS.ALL,
+        SEARCHCRITERIA.SYSTEM_NUMBER
+      );
+      if (vehiclesFromDB.length !== 1) {
+        throw ErrorHandler.Error(500, ERRORS.NO_UNIQUE_RECORD);
+      }
+      const uniqueRecord = vehiclesFromDB[0];
+
+      if (uniqueRecord.techRecord.filter(
+          (techRecord) => techRecord.statusCode === STATUS.PROVISIONAL
```

Comment:
leave it you but we could use composition with util functions here as we already check for that on line 187 `this.vehicle.techRecord[0].statusCode !== STATUS.PROVISIONAL`

so we could pass a status code when composing and flip the function above with !